cmake/kokkos_tribits.cmake

```diff
@@ -262,12 +272,20 @@ function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     target_compile_options(
       ${LIBRARY_NAME} PUBLIC $<$<COMPILE_LANGUAGE:${KOKKOS_COMPILE_LANGUAGE}>:${NODEDUP_CUDAFE_OPTIONS}>
     )
+
+    #exclude case of compiler_launcher forwarding to nvcc_wrapper as the used CXX compiler is shadowed in this case (compiler_launcher changes the compiler).
+    #The CXX compiler CMake will invoke for the check is not able to consume the cuda flags.
+    if(KOKKOS_ENABLE_COMPILE_AS_CMAKE_LANGUAGE OR ("${CMAKE_CXX_COMPILER}" MATCHES "nvcc_wrapper"))
```

Comment:
```suggestion
    if(KOKKOS_ENABLE_COMPILE_AS_CMAKE_LANGUAGE OR (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA))
```

Reply:
I tried this and it did not work. When automatically falling back to using the compiler_launcher, this would still try to invoke the CXX compiler which might not be able to consume the cuda arch flags.

Reply:
Yes, I misread the comment. The point is that `check_compiler_flag` always uses `CMAKE_CXX_COMPILER` and doesn't care about `CMAKE_CXX_COMPILER_LAUNCHER`.

Reply:
yes, correct